### PR TITLE
OMI_collider

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR adds OMI_collider. It supersedes #118. This PR takes #118, overhauls the text to be more readable, updates the table of equivalent types in major engines, adds some discussed features, fixes and completes the schema, and adds a folder of example GLTF files.

Preview: https://github.com/aaronfranke/gltf-extensions/tree/OMI_collider/extensions/2.0/OMI_collider

I have the document listed as a Stage 2 proposal. This PR meets all of the requirements of [Stage 1](https://github.com/omigroup/gltf-extensions#stage-1) and [Stage 2](https://github.com/omigroup/gltf-extensions#stage-2).

Stage 1

- [x] High level outline published as a GitHub PR in this repository. (this PR)
- [x] [Signed W3C CLA](https://www.w3.org/community/about/process/cla/). (yes)

Stage 2

- [x] Previous requirements from Stage 1. (yes)
- [x] One metaverse implementation of the extension in development. ([yes, in Godot Engine](https://github.com/godotengine/godot/pull/69266))
- [x] Valid and publicly accessible sample assets. (yes, see `examples/` subfolder)
- [x] Valid JSON schema describing the extension in its entirety. (yes, should be valid)
- [x] Markdown document explaining the purpose, features, and properties of the extension. (yes, see the `OMI_collider/README.md` file in this PR)

Stage 3

- [x] Previous requirements from Stage 2. (yes)
- [ ] Two or more metaverse implementations. (not yet, just Godot Engine)
- [x] glTF Validator implementation. ([yes, here's the pull request](https://github.com/KhronosGroup/glTF-Validator/pull/202))
- [x] Publicly available content creation tool for generating assets for this extension. (the Godot Engine implementation can export OMI_collider, although note that it will also export with OMI_physics_body)
- [ ] All implementations, sample assets, and content creation tools should consume or produce assets that pass validation. (awaiting more implementations)

### CODE REVIEW

**Comment by mrmetaverse** [comment]
Thanks for the demo yesterday (and last week too!) I can assign a reviewer and add this to the backlog to track on the main kanban. I know you had some questions about the process, and believe you all intend to discuss them during today's meeting, but feel free to update any changes in the comments.

File: extensions/2.0/OMI_collider/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+Can be optionally used together with the `OMI_physics_body` spec.
+
+## Overview
+
+Colliders can be added to glTF nodes along with information about the "type" of collider it is representing.
+
+This extension allows specifying shapes to be used in glTF scenes for collision or triggers. Shapes should be specified on their own glTF nodes.
+
+This extension does not describe what colliders should do within a scene, aside from whether they are a trigger or not. For a more thorough physics body specification, you need to also use the `OMI_physics_body` extension, and add `OMI_collider` nodes as children of the physics body. Multiple `OMI_collider` siblings with the same `isTrigger` setting may be treated as a single "compound collider" in game engines that support them.
```
Comment (RangerMauve) [inline extensions/2.0/OMI_collider/README.md:25]: Does the `isTrigger` wording here mean that you cannot use compound colliders as physics colliders? Mind elaborating on why if that's the case or rephrasing if it isn't?

**Review by RangerMauve — APPROVED** [review]
This is great thank you for picking this up!

File: extensions/2.0/OMI_collider/README.md (hunk 1 above)
Comment (aaronfranke) [reply extensions/2.0/OMI_collider/README.md:25]: You can use make compound colliders be either trigger or non-trigger. I'm not sure how else to rephrase it. If two `OMI_collider` children of a physics body have `isTrigger` set to false, they can be treated as a compound physics collider. If two `OMI_collider` children of a trigger body have `isTrigger` set to true, they can be treated as a compound trigger collider. The last two sentences can be simplified to "if `OMI_collider` siblings have the same `isTrigger` setting, they can be treated as a compound collider", which is basically what I wrote in the README.

**Comment by robertlong** [comment]
Vote to merge this PR passed with consensus
Screenshot 2023-01-05 at 3 15 28 PM

**Review by robertlong — APPROVED** [review]
Looks good!

**Pull request merged**